utils/dir/dir.go

```diff
@@ -0,0 +1,70 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package dir
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+// Params represent the input for CreateDataDir
+type Params struct {
+	Path      string
+	Perm      os.FileMode
+	Chown     bool
+	ChownPath string
```

Comment:
Why we need both Path and ChownPath?

We also should accept user name and group name

Reply:
> Why we need both Path and ChownPath?

ChownPath is the path of the directory whose user name and group name we want to apply to our newly created directory when we chown

I was using that because it keeps code like this contained in the `CreateDataDir` func

```go
        chownDirStat, err := os.Stat(prometheusDir)
	if err != nil {
		svc.l.Errorf("cannot get stat of %q: %v", prometheusDir, err)
	}

	chownDirSysStat := chownDirStat.Sys().(*syscall.Stat_t)
	uID, gID := int(chownDirSysStat.Uid), int(chownDirSysStat.Gid)
```

> We also should accept user name and group name

I think you meant user and group ID, changed the code to accept those instead of `ChownPath`

Reply:
No, we need names like "pmm", not numerical IDs. https://golang.org/pkg/os/user/ – there are looks function to get IDs by name

Reply:
done, thanks 😃